### ISSUE TITLE
nginx

### ISSUE THREAD

**Issue by Whoknowsit** [willnorris/imageproxy#59]
I'm currently trying to set up this great tool, using nginx, however, i must be too stupid.

I've installed imageproxy and started it:

`imageproxy -addr=127.0.0.1:8147`

I've created a simple nginx config:

```
server {
    listen 80;
    listen [::]:80;

    server_name     whatever-you-name-it.tld;

    access_log      /var/log/nginx/imageproxy.access.log;
    error_log       /var/log/nginx/imageproxy.error.log;

    location / {
        proxy_pass  http://127.0.0.1:8147;
        proxy_redirect off;
        break;
    }
}
```

Restarted nginx and done. But connecting to it via browser fails with "Empty response". So i've tried a Curl request to check, what's the issue:

`curl -o ~/test.jpg --user-agent "Mozilla 5.0" -H "Host: whatever-you-name-it.tld" 127.0.0.1:8080/https://willnorris.com/logo.jpg`

It worked. I had the test.jpg in the folder and it was not malformed, broken, etc.

So, the issue is not your script. However, i would like to ask if you have any idea, what's wrong on my end.

**Comment by willnorris** [comment]
You can see [my nginx config for willnorris.com](https://github.com/willnorris/willnorris.com/blob/7f4d3f3115c08c30f21e00e8c4b174d9927c5384/etc/nginx-shared.conf#L95-L104).  The pattern match was to prevent "http://" in the URL path from getting collapsed down to "http:/".  But if that was your problem, I would expect the curl command to be failing as well.  What happens if you run:

    curl -I -H "Host: whatever-you-name-it.tld" 127.0.0.1:8080/https://willnorris.com/logo.jpg

What are the full response headers?

**Comment by Whoknowsit** [comment]
Response:

```
HTTP/1.1 200 OK
Cache-Control: max-age=2764800
Content-Length: 8138
Content-Type: image/jpeg
Etag: "56bdf776-1fca"
Expires: Sun, 22 May 2016 07:10:53 GMT
Last-Modified: Fri, 12 Feb 2016 15:17:10 GMT
Date: Wed, 20 Apr 2016 16:13:45 GMT
```

Using the URL instead of ip:port, it returns

`curl: (52) Empty reply from server´

Strange... I mean... I have other services like gogs running on another subdomain, too and it works fine :D

**Comment by willnorris** [comment]
sounds like nginx is misconfigured for the different hosts that you serve.  If you're getting different responses when using the ip:port versus when using the host, then it doesn't sound like those are following the same path within nginx.  imageproxy knows nothing about the hostname used to reach the proxy; it only looks at the URL path.

**Comment by DenisIzmaylov** [comment]
@willnorris Thank you for your great project. But it would be better to add those solutions to README:

+ https://github.com/willnorris/imageproxy/issues/57#issuecomment-203972672
+ https://github.com/willnorris/imageproxy/issues/59#issuecomment-212492376

**Comment by willnorris** [comment]
Fixed by #62
